packages/core/src/test/shared/languageServer/lspManager.test.ts

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import * as sinon from 'sinon'
+import { LspManager, Manifest, Target } from '../../../shared/languageServer/lspManager'
+import { fs, makeTemporaryToolkitFolder } from '../../../shared'
+import * as download from '../../../shared/utilities/download'
+
+function createManifest(params?: {
+    deprecated?: boolean
+    versions?: {
+        serverVersion: string
+        isDelisted: boolean
+        targets: Target[]
+    }[]
+}): Manifest {
+    return {
+        manifestSchemaVersion: '1.0',
+        artifactId: 'test',
+        artifactDescription: 'test',
+        isManifestDeprecated: params?.deprecated ?? false,
+        versions: params?.versions ?? [],
+    }
+}
+
+describe('lspManager', () => {
+    let sandbox: sinon.SinonSandbox
+    let tempFolder: string
+
+    beforeEach(async () => {
+        sandbox = sinon.createSandbox()
+        tempFolder = await makeTemporaryToolkitFolder()
+    })
+
+    afterEach(async () => {
+        sandbox.restore()
+        await fs.delete(tempFolder)
+    })
+
+    describe('download', () => {
```

Comment:
These tests can likely be made obsolete by testing the same scenarios in tests for `AmazonQLSPInstaller`.
Stubbing the http fetcher should be enough to test the entire flow, and then you could have complex tests where it first downloads from the remote, then a subsequent call should instead use the cache

Reply:
yeah, the new tests are much simpler

Reply:
yeah now that the installer exists I can bring them up a level. I'll add that as a followup item